lesson35

shufladka opened the pull request.

shufladka committed:
lesson35

shufladka committed:
lesson35 // fix comments

shufladka committed:
lesson35 // fix troubles

gr00shik commented on lesson35/src/main/java/org/tms/entity/UserEntity.java:
```diff
@@ -0,0 +1,53 @@
+    // пополнение списка заданий пользователя
+    public void addNewTask(TaskEntity task) {
+
+        if (tasks == null || tasks.isEmpty()) {
```
tasks.isEmpty() - не нужно, если он пустой то он уже проинициализирован

gr00shik commented on lesson35/src/main/java/org/tms/service/impl/TaskServiceImpl.java:
```diff
@@ -0,0 +1,91 @@
+    }
+
+    @Override
+    public void getTaskByTaskId(String id) {
```
почему не сразу ожидать UUID?

gr00shik commented on lesson35/src/main/java/org/tms/service/impl/TaskServiceImpl.java:
```diff
@@ -0,0 +1,91 @@
+        Session session = HibernateConfig.create();
+        Transaction transaction = session.beginTransaction();
+
+        TaskEntity taskEntity = session.find(TaskEntity.class, userEntity.getUserId());
```
это как? мы ищем в таблице задач по ид юзера? это же не primary key

gr00shik commented on lesson35/src/main/java/org/tms/service/impl/UserServiceImpl.java:
```diff
@@ -0,0 +1,98 @@
+        Session session = HibernateConfig.create();
+        Transaction transaction = session.beginTransaction();
+
+        Query query = session.createQuery("update UserEntity as ue set ue.name =: nameUpdated, ue.dateOfBirth =: dateUpdated, ue.sex =: sexUpdated, ue.userType =: typeUpdated where ue.userId =: userEntity_uuid");
```
я бы на твоем месте просто достал бюы обьект а потом поменял поля, чем генерил большую строку

gr00shik reviewed (approved):
на доработку - см коммент по поводу ид

shufladka committed:
lesson35 // fixed:
1. tasks.isEmpty() is removed
2. 'String id' was changed to 'UUID uuid'
3. getTaskByUser(UserEntity userEntity) was removed

gr00shik reviewed (approved):
мержи

shufladka committed:
Merge branch 'main' into lesson35
